compiler/noirc_evaluator/src/ssa.rs

```diff
@@ -41,36 +41,56 @@ pub(crate) fn optimize_into_acir(
-        // Run mem2reg with the CFG separated into blocks
-        .run_pass(Ssa::mem2reg, "After Mem2Reg:")
-        .try_run_pass(Ssa::evaluate_assert_constant, "After Assert Constant:")?
-        .try_run_pass(Ssa::unroll_loops, "After Unrolling:")?
-        .run_pass(Ssa::simplify_cfg, "After Simplifying:")
-        .run_pass(Ssa::flatten_cfg, "After Flattening:")
-        .run_pass(Ssa::remove_bit_shifts, "After Removing Bit Shifts:")
-        // Run mem2reg once more with the flattened CFG to catch any remaining loads/stores
-        .run_pass(Ssa::mem2reg, "After Mem2Reg:")
-        .run_pass(Ssa::fold_constants, "After Constant Folding:")
-        .run_pass(Ssa::fold_constants_using_constraints, "After Constraint Folding:")
-        .run_pass(Ssa::dead_instruction_elimination, "After Dead Instruction Elimination:")
-        .finish();
+    let ssa =
+        SsaBuilder::new(program, print_ssa_passes, force_brillig_output, print_codegen_timings)?
+            .run_pass(Ssa::defunctionalize, "After Defunctionalization:")
+            .run_pass(Ssa::remove_paired_rc, "After Removing Paired rc_inc & rc_decs:")
+            .run_pass(Ssa::inline_functions, "After Inlining:")
+            // Run mem2reg with the CFG separated into blocks
+            .run_pass(Ssa::mem2reg, "After Mem2Reg:")
+            .try_run_pass(Ssa::evaluate_assert_constant, "After Assert Constant:")?
+            .try_run_pass(Ssa::unroll_loops, "After Unrolling:")?
+            .run_pass(Ssa::simplify_cfg, "After Simplifying:")
+            .run_pass(Ssa::flatten_cfg, "After Flattening:")
+            .run_pass(Ssa::remove_bit_shifts, "After Removing Bit Shifts:")
+            // Run mem2reg once more with the flattened CFG to catch any remaining loads/stores
+            .run_pass(Ssa::mem2reg, "After Mem2Reg:")
+            .run_pass(Ssa::fold_constants, "After Constant Folding:")
+            .run_pass(Ssa::fold_constants_using_constraints, "After Constraint Folding:")
+            .run_pass(Ssa::dead_instruction_elimination, "After Dead Instruction Elimination:")
+            .finish();
+
+    // For pretty printing the Brillig codegen time
+    let start_time = chrono::Utc::now().time();
 
     let brillig = ssa.to_brillig(print_brillig_trace);
 
+    if print_codegen_timings {
+        let end_time = chrono::Utc::now().time();
+        println!("SSA to Brillig: {} ms", (end_time - start_time).num_milliseconds());
+    }
+
     drop(ssa_gen_span_guard);
 
     let last_array_uses = ssa.find_last_array_uses();
 
-    ssa.into_acir(brillig, abi_distinctness, &last_array_uses)
+    // For pretty printing the ACIR codegen time
+    let start_time = chrono::Utc::now().time();
+
+    let acir = ssa.into_acir(brillig, abi_distinctness, &last_array_uses);
+
+    if print_codegen_timings {
+        let end_time = chrono::Utc::now().time();
+        println!("SSA to ACIR: {} ms", (end_time - start_time).num_milliseconds());
+    }
```

Comment:
Can you make a wrapper function here to time an action then print it?
It can be used here and for the to_brillig call.

Reply:
@vezenovm: https://github.com/noir-lang/noir/pull/4668

Reply:
Ah I see. Thanks